### PR TITLE
feat: 🎸 add basic server setup and logic

### PR DESCRIPTION
**Pull request by sj-williams**
- Service logic & tests

- Github build, test and standard CP workflows

relates to ministryofjustice/cloud-platform#6368

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+COPY . ./
+
+# Download all the dependencies
+RUN go mod download
+
+RUN CGO_ENABLED=0 go build -o /app/main .
+
+RUN chown -R appuser:appgroup /app
+
+FROM scratch
```
Comment (jaskaransarkaria) [inline Dockerfile:22]: ❤️

File: README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# cloud-platform-github-teams-filter
+
+## WIP
```
Comment (jaskaransarkaria) [inline README.md:3]: rename title

File: utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Strip github prefix
```
Comment (jaskaransarkaria) [inline utils/utils.go:8]: unnecessary comment

File: utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+func StripGithubPrefix(name string) string {
+	return strings.Replace(name, "github:", "", -1)
+}
+
+// Deduplicate teams
```
Comment (jaskaransarkaria) [inline utils/utils.go:13]: unnecessary comment

File: utils/utils.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+		if !seen[team] {
+			seen[team] = true
+			dedupTeams = append(dedupTeams, team)
+		}
+	}
+
+	return dedupTeams
+}
+
+// Filter out teams that are not present in the cluster teams slice
```
Comment (jaskaransarkaria) [inline utils/utils.go:28]: unnecessary comment, maybe rename func to be more descriptive?

File: utils/utils.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+func FilterTeams(input string, cluster []string) string {
+
+	// Convert input teams string to slice
```
Comment (jaskaransarkaria) [inline utils/utils.go:31]: unnecessary comment

File: utils/utils.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	inputSlice := strings.Split(input, ":")
+	fmt.Println(inputSlice)
```
Comment (jaskaransarkaria) [inline utils/utils.go:33]: do we still need a log here?

File: utils/utils.go [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	var result []string
+	for _, team := range inputSlice {
+		if clusterMap[team] {
+			result = append(result, team)
+		}
+	}
+
+	// Convert filtered slice back to string
```
Comment (jaskaransarkaria) [inline utils/utils.go:48]: unnecessary comment

File: utils/utils_test.go [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+import "testing"
+
+func TestStripGithubPrefix(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
```
Comment (jaskaransarkaria) [inline utils/utils_test.go:11]: Is it worth adding a test case for if there's no `github:` prefix? Or a malformed one?

File: utils/utils_test.go [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+	}
+}
+
+func TestRemoveDuplicates(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []string
+		expected []string
+	}{
+		{
```
Comment (jaskaransarkaria) [inline utils/utils_test.go:39]: add test case for no duplicates?

File: utils/utils_test.go [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+func TestFilterTeams(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		cluster  []string
+		expected string
+	}{
+		{
```
Comment (jaskaransarkaria) [inline utils/utils_test.go:68]: test case for empty string input?

**Review by jaskaransarkaria — COMMENTED** [review]
Love love love how clean and easy to read this repo is ✨

File: init_app/init_env_vars.go [hunk 12]
```diff
@@ -0,0 +1,16 @@
+package init_app
+
+import (
+	"os"
+)
+
+func InitEnvVars() string {
+	ginMode := "debug"
```
Comment (mikebell) [inline init_app/init_env_vars.go:8]: Just checking you still want to run this in debug?

Comment (jaskaransarkaria) [reply init_app/init_env_vars.go:8]: This bit of the code is a bit misleading (my bad it's adopted from label-pods) but basically it defaults to debug if there's now value set otherwise it will use the value set in the env var

File: utils/utils_test.go (hunk 9 above)
Comment (sj-williams) [reply utils/utils_test.go:11]: not sure if thats out of scope, maybe we should be doing a conftest for user PRs on that though?

**Review by jaskaransarkaria — APPROVED** [review]
LOVE IT 🎉
